src/applications/check-in/components/pages/Appointments/index.jsx

```diff
@@ -57,6 +63,44 @@ const AppointmentsPage = props => {
     [checkInDataError, updateError, app],
   );
 
+  useEffect(
+    () => {
+      if (refresh) {
+        refreshCheckInData();
+        setLoadedAppointments([]);
+        setRefresh(false);
+      }
+    },
+    [refresh, refreshCheckInData],
+  );
+
+  useEffect(
```

Comment:
Should there be a setRefresh(true) in here somewhere?

Reply:
Is there also a way to test this with e2e?

Reply:
I decided that the auto refresh should just hit check-in data, since its not fired from the button and not user initiated, and it is a result of a value in the vetext data only.

Unfortunately, there is not a great way to e2e this because cypress doesn't like waiting around for stuff and it could potentially drag the entire e2e process. I spent a couple hours trying to think around this. Do you have any ideas?

Reply:
I agree with that reasoning. Since the flow of the app is different and they have to get through all the questions maybe should we make the 5 seconds longer?

For e2e what if the check in window end was less than 5 seconds away from date.now()

Reply:
The 5 seconds isn't there to give them time to complete check-in, its there to make sure they don't attempt it if the appointment time will be elapsed. So the intent is to fetch the new eligibility which should then remove the button.

Want to co-work on the e2e stuff? I couldn't get a reliable test to work even with a window in spec. Its still working on timeouts which didn't work consistently